cli/src/main/scala/laserdisc/cli/CLI.scala

```diff
@@ -60,90 +61,91 @@ object CLI extends IOApp.WithContext { self =>
+        case (Some(ip), Some(port)) =>
+          IO(println(logo)) >> impl.mkStream(ip, port)(SyncLogWriter.noOpLogF[IO]).compile.drain.as(ExitCode.Success)
+        case _ => IO(println("please supply valid host and port (space separated)")).as(ExitCode.Error)
       }
 
     case _ => IO(println("please supply host and port (space separated)")).as(ExitCode.Error)
   }
 
   private[cli] final object impl {
+
     private[this] val tb = universe.runtimeMirror(self.getClass.getClassLoader).mkToolBox()
 
-    def mkStream(host: Host, port: Port): Stream[IO, ExitCode] =
-      Stream.resource(Blocker.fromExecutorService[IO](IO(fromExecutorService(newSingleThreadExecutor())))) >>= { replBlockingEC =>
-        Fs2LogWriter.consoleLogStream[IO] >>= { implicit log =>
-          RedisClient.toNode[IO](host, port) evalMap { redisClient =>
-            val promptStream: Stream[IO, String] = Stream.emit(s"$host:$port> ").repeat
-
-            val emptyPrompt: IO[Unit] =
-              promptStream.head.through(text.utf8Encode).through(io.stdout(replBlockingEC)).compile.drain
-
-            def prompt(msg: String): IO[Unit] =
-              Stream
-                .emit(msg)
-                .append(promptStream.head)
-                .through(text.utf8Encode)
-                .through(io.stdout(replBlockingEC))
-                .compile
-                .drain
-
-            final object ToMillis {
-              def unapply(nanos: Long): Option[Double] = Some(nanos.toDouble / 1000000d)
-            }
+    def mkStream(host: Host, port: Port)(implicit `_`: LogWriter[IO]): Stream[IO, ExitCode] =
+      Stream.resource(Blocker.fromExecutorService[IO](IO(fromExecutorService(newSingleThreadExecutor)))) >>= { replBlockingPool =>
+        Stream.resource(RedisClient.toNode[IO](host, port)) evalMap { redisClient =>
+          val promptStream: Stream[IO, String] = Stream.emit(s"$host:$port> ").repeat
+
+          val emptyPrompt: IO[Unit] =
+            promptStream.head
+              .through(text.utf8Encode)
+              .through(io.stdout(replBlockingPool))
+              .compile
+              .drain
+
+          def prompt(msg: String): IO[Unit] =
+            Stream
+              .emit(msg)
+              .append(promptStream.head)
+              .through(text.utf8Encode)
+              .through(io.stdout(replBlockingPool))
+              .compile
+              .drain
+
+          final object ToMillis {
+            def unapply(nanos: Long): Option[Double] = Some(nanos.toDouble / 1000000d)
+          }
 
-            val protocol: Pipe[IO, String, Protocol] = {
-              def mkProtocol(code: String): IO[Maybe[Protocol]] =
-                IO.delay {
-                  tb.eval {
-                      tb.parse {
-                        s"""import laserdisc._
-                           |import laserdisc.auto._
-                           |import laserdisc.all._
```

Comment:
i think we need this for now (see [here](https://github.com/laserdisc-io/laserdisc/blob/master/core/src/main/scala/laserdisc/syntax.scala#L17)) or we lose the ability to just invoke all protocols w/o prefix in the CLI

Reply:
I think the protocols in `all` are already imported by `laserdisc._`. If I add it and run
```
localhost:6379> strings.set("a", 23)
```
I get
```
reflective compilation has failed:

reference to strings is ambiguous;
it is imported twice in the same scope by
import laserdisc.all._
and import laserdisc._
localhost:6379>
```
The one that doesn't seem to be found is `BListP` though.

Reply:
Yeah, in theory it should be just `set(“a”, 23)`

`strings` is another object that extends `StringP` (Like `all`, which extends all protocols)